src/Microsoft.TemplateEngine.Cli/HelpAndUsage/HelpForTemplateResolution.cs

```diff
@@ -423,51 +399,84 @@ public static CreationResultStatus HandleParseError(INewCommandInput commandInpu
             return CreationResultStatus.InvalidParamValues;
         }
 
-        private static string GetLanguageMismatchErrorMessage(INewCommandInput commandInput)
+        private static string GetInputParametersString(string templateName, string templateLanguage, string context, string baselineName)
         {
-            string inputFlagForm;
-            if (commandInput.Tokens.Contains("-lang"))
-            {
-                inputFlagForm = "-lang";
-            }
-            else
-            {
-                inputFlagForm = "--language";
-            }
+            StringBuilder inputParametersList = new StringBuilder();
+            string separator = ", ";
 
-            string invalidLanguageErrorText = LocalizableStrings.InvalidTemplateParameterValues;
-            invalidLanguageErrorText += Environment.NewLine + string.Format(LocalizableStrings.InvalidParameterDetail, inputFlagForm, commandInput.Language, "language");
-            return invalidLanguageErrorText;
-        }
-
-        private static string GetInputParametersString(string templateName, string templateLanguage, string context)
-        {
-            List<string> inputParametersList = new List<string>();
             if (!string.IsNullOrEmpty(templateName)){
-                inputParametersList.Add("'" + templateName + "'");
+                inputParametersList.AppendFormat("'{0}'", templateName);
             }
-            if (!string.IsNullOrEmpty(templateLanguage)){
-                inputParametersList.Add("lang=" + templateLanguage);
+            if (!string.IsNullOrEmpty(templateLanguage))
+            {
+                if (inputParametersList.Length != 0)
```

Comment:
I'm in doubt here. This possibly get code more readable, however since all parameters are optional we will need additional check so it results in one of below:
`string.IsNullOrEmpty(templateName) ? inputParametersString.ToString().Substring(separator.Length) : inputParametersString.ToString();`
`inputParametersString.Length == 0 ? string.Empty : inputParametersString.ToString().Substring(separator.Length);`
Also performance wise it would be worse.

Reply:
Yes, the check in the end will be needed.

You are right that if you call `ToString().Substring(...)`, that will perform worse. Because you are creating two strings. But StringBuilder can directly do that for you if you just call `ToString(...)`.

 Modifying your second option, this would probably work:
```csharp
return inputParametersString.Length == 0 ? string.Empty : inputParametersString.ToString(separator.Length);
```

StringBuilder copies your data into [a new memory](https://referencesource.microsoft.com/#mscorlib/system/text/stringbuilder.cs,338) section in any case. So specifying a range shouldn't be any worse in performance. And you save yourself a couple of if checks.

This is more of a cosmetic change and not a "must do", so it is up to you.

Reply:
ok, I was not aware that StringBuilder has actual "substring" ToString() option, nice catch.  Will use that.